Tools/include/Tools/HgcrocEmulator.h

```diff
@@ -156,7 +156,9 @@ class HgcrocEmulator {
   bool digitize(
       const int& channelID, const std::vector<double>& voltages,
       const std::vector<double>& times,
-      std::vector<ldmx::HgcrocDigiCollection::Sample>& digiToAdd) const;
+      std::vector<ldmx::HgcrocDigiCollection::Sample>& digiToAdd,
+      const double &attenuation=0.,                                                                                                                                                           
+      const double &shift=0.) const;
```

Comment:
These are smart additions to the emulator. Could you give a comment in the docs-comment about what they are for? Specifically, units are especially important.

Reply:
So, I'm a little concerned here.  The HCAL bars are big and we could have hits at quite different times, when taking into account the light-propagation effects.  Is this to be called only for 'voltages' within a small region in space along the bar?  I guess these 'voltages' are really PE or something similar?

Reply:
You are correct. The 'voltages' are just unit-converted individual sim hits. I've called them 'voltages' to emphasize that the units should be mV and not fC or MeV.

Reply:
@jmmans and I talked about the best way to move forward and I believe this is in two stages.

### 1. Remove the attenuation and shift from this digitize function.
Without changing the HgcrocEmulator on `trunk`, you would get the same affect as what you have on this branch now if you factor in the attenuation and shift you calculated *onto each hit individually*.

This is important because, particularly for the HCal, hits can occur at widely-separated positions or times (but in the same bar), so their attenuations and shifts would be wildly different. 

Long story short, the voltages/times input into the `digitize` function should the voltages/times **at the chip** and should be calculated **for each sim hit individually**.

### 2. Update HgcrocEmulator to handle multiple hits better
The shortcut that is currently in the emulator works well for single hits in each cell/bar or for multiple hits that are near in time and space. This shortcut works for a majority of digitizations in both the ECal and HCal, but is not realistic.

@jmmans and I will work together on improving the emulator to handle multiple hits appropriately (Issue #946 ). Also, if the HCal calculates the attenuation and shift *for each sim-hit individually*, then the HCal would not need to change their code.

Reply:
Alright, I think this makes sense. I have included these modifications [here](https://github.com/LDMX-Software/ldmx-sw/pull/958/commits/49d1a6bdbe8242520b52a9782e981f100e57fc6f) and [here for Hcal](https://github.com/LDMX-Software/Hcal/commit/4c9b4b80fe20ba75b9bd81690b0039bbc6068a8c)